File: cwt-spring/src/main/java/com/cwtsite/cwt/configuration/entity/Configuration.java
package com.cwtsite.cwt.configuration.entity;


import com.cwtsite.cwt.configuration.entity.enumeratuion.ConfigurationKey;
import com.cwtsite.cwt.user.repository.entity.User;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Timestamp;

@Entity
@Table(name = "configuration")
public class Configuration implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "key")
    @Enumerated(EnumType.STRING)
    private ConfigurationKey key;

    @Column(name = "value", columnDefinition = "text")
    private String value;

    @Column(name = "modified")
    @UpdateTimestamp
    private Timestamp modified;

    @ManyToOne(optional = false)
    private User author;

    public ConfigurationKey getKey() {
        return key;
    }

    public void setKey(ConfigurationKey key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public Timestamp getModified() {
        return modified;
    }

    public void setModified(Timestamp modified) {
        this.modified = modified;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }
}

File: cwt-spring/src/main/java/com/cwtsite/cwt/domain/configuration/entity/Configuration.kt
package com.cwtsite.cwt.domain.configuration.entity

import com.cwtsite.cwt.domain.configuration.entity.enumeratuion.ConfigurationKey
import com.cwtsite.cwt.domain.user.repository.entity.User
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.FetchType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "configuration")
data class Configuration(

    @Id
    @Column(name = "key")
    @Enumerated(EnumType.STRING)
    var key: ConfigurationKey,

    @Column(name = "value", columnDefinition = "text")
    var value: String?,

    @Column(name = "description", columnDefinition = "text")
    var description: String? = null,

    @Column(name = "modified")
    @field:UpdateTimestamp
    var modified: Instant? = null,

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(nullable = false)
    var author: User? = null
) {
    override fun toString(): String {
        return "Configuration{id=$key, value=$value}"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Configuration

        if (key != other.key) return false

        return true
    }

    override fun hashCode(): Int {
        return key.hashCode()
    }
}
